src/error.rs

```diff
@@ -100,6 +100,12 @@ impl From<Context<HandlerErrorKind>> for HandlerError {
     }
 }
 
+impl From<Error> for HandlerError {
```

Comment:
This conversion from any failure Error is maybe too lax for HandlerError, I think

If you return a regular Result instead of HandlerResult from Server::start (it's not a Handler anyway so it doesn't make sense there) this conversion wouldn't be needed